webpack/scenes/ContentViews/Publish/CVPublishFinish.js

```diff
@@ -0,0 +1,147 @@
+  const status = useSelector(state => selectPublishContentViewStatus(state, cvId, versionCount));
+  const error = useSelector(state => selectPublishContentViewsError(state, cvId, versionCount));
+  const pollResponse = useSelector(state =>
+    selectPublishTaskPoll(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+  const pollResponseStatus = useSelector(state =>
+    selectPublishTaskPollStatus(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+
+  useEffect(() => {
+    setSaving(true);
+    dispatch(publishContentView({
+      id: cvId,
+      versionCount,
+      description,
+      environment_ids: checkedItems.map(item => item.id),
+      is_force_promote: (forcePromote.length > 0),
+    }));
+    setDescription('');
+    setCheckedItems([]);
+  }, [dispatch, setSaving, setDescription, setCheckedItems]);
+
+  useEffect(() => {
+    setSaving(true);
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setSaving(false);
+      dispatch(pollPublishTask(cvVersionPublishKey(cvId, versionCount), response));
+      setPolling(true);
+    } else if (status === STATUS.ERROR) {
+      setSaving(false);
+    }
+  }, [JSON.stringify(response), status, error]);
+
+  useEffect(() => {
+    const { state, result } = pollResponse;
+    if (response && (state === 'paused' || result === 'error')) {
+      setTaskErrored(true);
+    }
+    if (response && (state === 'stopped' && result === 'success')) {
+      setPolling(false);
+      dispatch(handleFinishedTask(cvId, cvVersionPublishKey(cvId, versionCount), pollResponse));
+      dispatch(getContentViewDetails(cvId));
+      dispatch(getContentViews);
+      setIsOpen(false);
+      setRedirect(true);
+    }
+  }, [JSON.stringify(pollResponse), dispatch, setTaskErrored,
+    setPolling, setIsOpen, setRedirect, pollResponseStatus]);
+
+  if (saving) {
+    return <Loading />;
+  }
+
+  if (polling && pollResponse) {
+    return (
+      <Stack hasGutter>
+        <StackItem span={12} rowSpan={3}>
+          <Bullseye>
+            <div style={{ height: '350px', width: '350px' }}>
+              <ChartDonut
+                ariaDesc="Publishing Content View"
+                ariaTitle="Publishing Content View"
+                data={[{ x: 'Completed', y: pollResponse.progress * 100 || 0 }, { x: 'Pending', y: (100 - (pollResponse.progress * 100)) || 100 }]}
+                labels={({ datum }) => `${datum.x}: ${datum.y}%`}
+                title="Publishing"
+                colorScale={taskErrored ? ['#FF0000', '#F08080'] : ['#008000', '#8FBC8F']}
+              />
+            </div>
+          </Bullseye>
+        </StackItem>
+        <StackItem isFilled> </StackItem>
+        <StackItem span={6}>
+          <Bullseye>
+            <Button onClick={() => { dispatch(handleFinishedTask(cvId, cvVersionPublishKey(cvId, versionCount), pollResponse, true)); setIsOpen(false); }} variant="primary" aria-label="publish_content_view">
```

Comment:
These buttons kinda look like they're in the center of the screen - there's too much space between them and the bottom of the modal.  Should they be moved down a bit?

Reply:
Increased size of status donut chart and buttons are down below.
![Screenshot from 2021-06-14 22-37-00](https://user-images.githubusercontent.com/21146741/121984501-10a98900-cd61-11eb-961a-0ecb1a49bfe6.png)

Reply:
this looks much better! :+1:

Reply:
```js
() => { dispatch(handleFinishedTask(cvId, cvVersionPublishKey(cvId, versionCount), pollResponse, true)); setIsOpen(false);
```
This is multiple lines of code; probably better to extract into a named function at this point.